include/deal.II/multigrid/multigrid.templates.h

```diff
@@ -215,72 +168,40 @@ template <typename VectorType>
 void
 Multigrid<VectorType>::level_step(const unsigned int level, Cycle cycle)
 {
-  char cychar = '?';
-  switch (cycle)
-    {
-      case v_cycle:
-        cychar = 'V';
-        break;
-      case f_cycle:
-        cychar = 'F';
-        break;
-      case w_cycle:
-        cychar = 'W';
-        break;
-      default:
-        Assert(false, ExcNotImplemented());
-    }
-
-  if (debug > 0)
-    deallog << cychar << "-cycle entering level  " << level << std::endl;
-
   // Combine the defect from the initial copy_to_mg with the one that has come
   // from the finer level by the transfer
   defect2[level] += defect[level];
   defect[level] = typename VectorType::value_type(0.);
 
-  if (debug > 2)
-    deallog << cychar << "-cycle defect norm     " << defect2[level].l2_norm()
-            << std::endl;
-
   if (level == minlevel)
     {
-      if (debug > 0)
-        deallog << cychar << "-cycle coarse level    " << level << std::endl;
-
+      this->signals.coarse_solve(true, level);
       (*coarse)(level, solution[level], defect2[level]);
+      this->signals.coarse_solve(false, level);
```

Comment:
Had we just forgotten to call all of these signals? They appear to have existed, after all...

Reply:
yes, I think I only tested it for the V cycle before.

Reply:
Yes, we treat the V-cycle in `level_v_step` separately and only called the signals there.